analyzers/tests/SonarAnalyzer.UnitTest/TestCases/GetTypeWithIsAssignableFrom.cs

```diff
@@ -92,6 +92,33 @@ public void Go(Repro_3605 value)
         }
     }
 
+    // https://github.com/SonarSource/sonar-dotnet/issues/6616
+    public class Repro_6616
+    {
+        public void IsInstanceOfType(object obj)
+        {
+            _ = typeof(ISet<int>).IsInstanceOfType(obj); // Noncompliant, bounded generic type
+            _ = typeof(ISet<>).IsInstanceOfType(obj); // Compliant, unbonded generic type
+            _ = typeof(IDictionary<int, int>).IsInstanceOfType(obj); // Noncompliant, bounded generic type
+            _ = typeof(IDictionary<,>).IsInstanceOfType(obj); // Compliant, unbonded generic type
+
+            Type t = typeof(ISet<>);
```

Comment:
We don't track the value in any way, are we?  So an method parameter `Type type` should do the trick as well

Reply:
Having `Type t1 = typeof(ISet<>)` was meant to show that, indeed, the rule stays at syntactical level, and doesn't do any inference about the possible value of `t1` when analyzing `t1.IsAssignableFrom(t2)`.

I agree the test case conflates two tests in one: not a `typeof` expression + lack of "value tracking", when the value would actually be a `typeof`.

What about having both, and being more explicit about what the test is about:
```cs
            _ = tPar.IsInstanceOfType(obj); // Compliant, not a typeof expression

            tPar = typeof(ISet<>);
            _ = t.IsInstanceOfType(obj); // Compliant, not a typeof expression and value not tracked
```

Reply:
Sure, why not. More UTs is always good.